### PR TITLE
Convert YAMLTextArea to code mirror editor

### PR DESCRIPTION
**Pull request by bramkragten**
yaml editor

### CODE REVIEW

File: src/components/ha-yaml-editor.ts [hunk 1]
```diff
@@ -17,12 +17,11 @@ declare global {
...
-@customElement("hui-yaml-editor")
-export class HuiYamlEditor extends HTMLElement {
-  public _hass?: HomeAssistant;
-
+@customElement("ha-yaml-editor")
+export class HaYamlEditor extends HTMLElement {
   public codemirror!: any;
-
+  public autofocus = false;
+  private _hass?: HomeAssistant;
```
Comment (balloob) [inline src/components/ha-yaml-editor.ts]: If we need this just for RTL, we might want to consider to just pass that in. That way it's not so tied into HA.

File: src/panels/config/js/yaml_textarea.tsx [hunk 2]
```diff
@@ -55,26 +58,39 @@ export default class YAMLTextArea extends Component<any, any> {
       isValid,
     });
     if (isValid) {
+      this.yamlEditor.style.removeProperty(
```
Comment (balloob) [inline src/panels/config/js/yaml_textarea.tsx]: Any reason we don't add this to the editor itself ?

File: src/panels/config/js/yaml_textarea.tsx [hunk 3]
```diff
@@ -55,26 +58,39 @@ export default class YAMLTextArea extends Component<any, any> {
...
-        label={label}
-        value={value}
-        style={style}
-        onvalue-changed={this.onChange}
-        dir="ltr"
-      />
+      <div>
+        <p>{label}</p>
+        <ha-yaml-editor
+          ref={(yamlEditor) => (this.yamlEditor = yamlEditor)}
```
Comment (balloob) [inline src/panels/config/js/yaml_textarea.tsx]: We shouldn't declare ref functions inline. Use a class function instead.

**Review by balloob — APPROVED** [review]
Amazing !

File: src/components/ha-yaml-editor.ts [hunk 4]
```diff
@@ -82,12 +76,18 @@ export class HaYamlEditor extends HTMLElement {
...
+    this.classList.toggle("error-state", error);
+  }
+
   get hasComments(): boolean {
     return this.shadowRoot!.querySelector("span.cm-comment") ? true : false;
   }
 
   public connectedCallback(): void {
     if (!this.codemirror) {
+      console.log(this.rtl);
```
Comment (balloob) [inline src/components/ha-yaml-editor.ts]: console.log

Comment (bramkragten) [reply src/components/ha-yaml-editor.ts]: Outdated

**Pull request merged**